advanced/Scripts/updatecheck.sh

```diff
@@ -41,54 +41,78 @@ function get_local_version() {
 # shellcheck disable=SC1091
 . /etc/pihole/setupVars.conf
 
+utilsfile="/opt/pihole/utils.sh"
+source "${utilsfile}"
+
+# Remove the below three legacy files if they exist
+GITHUB_VERSION_FILE="/etc/pihole/GitHubVersions"
+LOCAL_BRANCH_FILE="/etc/pihole/localbranches"
+LOCAL_VERSION_FILE="/etc/pihole/localversions"
+
+if [ -f "${GITHUB_VERSION_FILE}" ]; then
+    rm "${GITHUB_VERSION_FILE}"
+fi
+
+if [ -f "${LOCAL_BRANCH_FILE}" ]; then
+    rm "${LOCAL_BRANCH_FILE}"
+fi
+
+if [ -f "${LOCAL_VERSION_FILE}" ]; then
+    rm "${LOCAL_VERSION_FILE}"
+fi
+
+# Create new versions file if it does not exist
+VERSION_FILE="/etc/pihole/versions"
+
+if [ ! -f "${VERSION_FILE}" ]; then
+    touch "${VERSION_FILE}"
+    chmod 644 "${VERSION_FILE}"
```

Comment:
It should be safe to `touch` and `chmod` the file always. `touch` will just modify the timestamps if the file already exists (it won't change the content). And `chmod` is safe to do anytime.